api/v1alpha1/hostedclusterinfrastructure_types.go

```diff
@@ -0,0 +1,344 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	Verified                 = "Verified"
+	AsExpected               = "AsExpected"
+	NotAsExpected            = "NotAsExpected"
+	MisConfiguredReason      = "MisConfigured"
+	RemovingReason           = "Removing"
+	PlatfromDestroyReason    = "Destroying"
+	HCIBeingConfiguredReason = "BeingConfigured"
+
+	// ProviderSecretConfigured indicates the state of the secret reference
+	ProviderSecretConfigured ConditionType = "ProviderSecretConfigured"
+
+	// PlatformConfigured indicates (if status is true) that the
+	// platform configuration specified for the platform provider has been applied
+	PlatformConfigured ConditionType = "PlatformInfrastructureConfigured"
+
+	// PlatformIAMConfigured indicates (if status is true) that the IAM is configured
+	PlatformIAMConfigured ConditionType = "PlatformIAMConfigured"
+)
+
+func init() {
+	SchemeBuilder.Register(&HostedClusterInfrastructure{}, &HostedClusterInfrastructureList{})
+}
+
+// HostedClusterInfrastructureSpec is the desired behavior of a HostedClusterInfrastructure.
+type HostedClusterInfrastructureSpec struct {
+
+	// InfraID is a globally unique identifier for the cluster. This identifier
+	// will be used to associate various cloud resources with the HostedCluster
+	// and its associated NodePools. If not specified the metadata.name for this
+	// resource is used. When specified, this value is used.
+	//
+	// +optional
+	// +immutable
+	InfraID string `json:"infraID,omitempty"`
+
+	// SubDomain is the identifier that is used between the *.apps and base-domain
+	// when building an ingress URL. This will also be the name of the cluster.
+	// If no value is provided, the infra-id is used
+	// When SubDomain = my and base-domain = company.com, the resulting ingress URL
+	// will be *.apps.my.company.com
+	//
+	// +optional
+	// +immutable
+	SubDomain string `json:"subDomain,omitempty"`
+
+	// Platform specifies the underlying infrastructure provider for the cluster
+	// and is used to configure platform specific behavior.
+	//
+	// +immutable
+	Platform PlatformInfraSpec `json:"platform"`
+
+	// baseDomain is the base domain of the cluster. All managed DNS records will
+	// be sub-domains of this base.
+	//
+	// For example, given the base domain `openshift.example.com`, an API server
+	// DNS record may be created for `cluster-api.openshift.example.com`.
+	//
+	// Once set, this field cannot be changed.
+	BaseDomain string `json:"baseDomain"`
+
+	// CloudProvider secret, contains the Cloud credenetial and Base Domain
```

Comment:
typo "credenetial".

Which format should I use to fill in this secret?
Which permissions should I give to whatever I ref here?

Reply:
This is a Cloud Provider credential so its used to store the AWS credential for the controller to use. This should be usable with MCE's default AWS Cloud provider credential as well.  The CLI makes the secret (its shown in the render) for better or worse.